docs/design/architecture.md

```diff
@@ -2,128 +2,617 @@
+
+> **Note:**
+>
+> For the purposes of this document and to keep explanations
+> simpler, we assume the user is running this command in the
+> [host environment](#environments).
+
+## Container image
+
+In the [example command](#example-command) the user has specified the
+type of container they wish to run via the container image name:
+`ubuntu`. This image name corresponds to a "container image" that can
+be used to create a container with an Ubuntu Linux environment. Hence,
+in our [example](#example-command), the `sh(1)` command will be run
+inside a container which has an Ubuntu rootfs.
+
+> **Note:**
+>
+> The term "container image" is confusing since the image in question
+> is **not** a container: it is simply a set of files ("an image")
+> that can be used to _create_ a container. The term "container
+> template" would be more accurate but the term "container image" is
+> commonly used so this document uses the standard term.
+
+For the purposes of this document, the most important part of the
+command above is the container image the user has requested. Normally,
+a container image will be "pulled" (downloaded) from a remote site and
+a copy stored locally. This local container image is used by the
+container manager to create an [OCI bundle](#oci-bundle) which will
+form the environment the container will run in.
+
+containerd itself does not create containers, it simply manages them.
+However, before it launches a [runtime](#runtime) to create the
+container, containerd will create an [OCI rootfs bundle](https://github.com/opencontainers/runtime-spec/blob/main/bundle.md).
+
+## OCI bundle
+
+To understand what follows, it is important to know at a high level
+how an OCI compatible container is created.
+
+An OCI compatible container is created by taking a
+[container image](#container-image) and converting the embedded rootfs
+into an
+[OCI rootfs bundle](https://github.com/opencontainers/runtime-spec/blob/main/bundle.md),
+or more simply, an _OCI bundle_.
+
+An OCI bundle is a `tar(1)` archive normally created by a container
+manager which is passed to an OCI [runtime](#runtime) which converts
+it into a full container rootfs. The bundle contains two assets:
+
+- A container image "rootfs"
+
+  This is simply a directory of files that will be used to represent
+  the rootfs for the container.
+
+  For the [example command](#example-command), the directory will
+  contain the files necessary to create a minimal Ubuntu root
+  filesystem.
+
+- An [OCI configuration file](https://github.com/opencontainers/runtime-spec/blob/main/config.md)
+
+  This is a JSON file called `config.json`.
+
+  The container manager will create this file so that:
+
+  - The `root.path` value is set to the full path of the specified
+    container rootfs.
+
+    In [our example](#example-command) this value will be `ubuntu`.
+
+  - The `process.args` array to the list of commands the user wishes
```

Comment:
Did you mean to say " the process.args array contains the list of commands the user wishes..."